### PR TITLE
[skip-ci] Remove proxy setting and upgrade to Ubuntu 20.04 for SOF docker images

### PR DESCRIPTION
**Pull request by fredoh9**
fixes: https://github.com/thesofproject/sof/issues/3831

### CODE REVIEW

**Comment by fredoh9** [comment]
1. Removed apt.conf completely.
2. Upgrade base Ubuntu image to 20.04

Tested with proxy environment and non-proxy environment.
Docker images are ready to switch

- thesofproject/sof:20210511
- thesofproject/sofqemu:20210511

If I get approval, I will switch to latest for build machines.

File: scripts/docker-run.sh [hunk 1]
```diff
@@ -19,6 +19,8 @@ fi
 docker run -i -v "$(pwd)":/home/sof/work/sof.git \
 	--env CMAKE_BUILD_TYPE \
 	--env PRIVATE_KEY_OPTION \
-	   --user "$(id -u)" \
+	--env http_proxy=${http_proxy} \
+	--env https_proxy=${https_proxy} \
```
Comment (marc-hb) [inline scripts/docker-run.sh:23]: Something tells me you haven't run shellcheck :-)

File: scripts/docker_build/sof_builder/Dockerfile [hunk 2]
```diff
@@ -14,19 +14,23 @@
...
 
+# for non-interactive package install
+ARG DEBIAN_FRONTEND=noninteractive
+
 RUN apt-get -y update && \
 	    apt-get install -y \
-	    	autoconf \
+		dialog \
+		apt-utils \
+		vim \
```
Comment (marc-hb) [inline scripts/docker_build/sof_builder/Dockerfile:32]: Can I have emacs too? Just kidding, however `jed` would be nice.

File: scripts/docker_build/sof_builder/Dockerfile [hunk 3]
```diff
@@ -63,7 +67,7 @@ ENV HOME /home/sof
 RUN mkdir -p /home/sof/work/alsa && cd /home/sof/work/alsa && \
 git clone $CLONE_DEFAULTS https://github.com/thesofproject/alsa-lib.git && \
 git clone $CLONE_DEFAULTS https://github.com/thesofproject/alsa-utils.git && \
-cd /home/sof/work/alsa/alsa-lib && ./gitcompile &&  make install && \
+cd /home/sof/work/alsa/alsa-lib && ./gitcompile --prefix=/usr --libdir=/usr/lib/x86_64-linux-gnu/ &&  make install && \
```
Comment (marc-hb) [inline scripts/docker_build/sof_builder/Dockerfile:70]: Is this related to #2543? Some other bug? Please add a comment or URL.

**Review by marc-hb — COMMENTED** [review]
LGTM except `-prefix=/usr --libdir=/usr/lib/x86_64-linux-gnu/` is missing a comment or URL

**Comment by marc-hb** [comment]
Forgot this sorry: can you please rename this PR? It's far more than proxy settings now.

File: scripts/docker_build/sof_qemu/Dockerfile [hunk 4]
```diff
@@ -22,8 +22,13 @@ ARG host_https_proxy
 ENV http_proxy $host_http_proxy
 ENV https_proxy $host_https_proxy
```
Comment (marc-hb) [inline scripts/docker_build/sof_qemu/Dockerfile:23]: remove for QEMU image too?

Comment (marc-hb) [reply scripts/docker_build/sof_qemu/Dockerfile:23]: Share some of the code to avoid copy/paste/diverge?

File: scripts/docker-run.sh (hunk 1 above)
Comment (fredoh9) [reply scripts/docker-run.sh]: shellcheck saves a live!

File: scripts/docker_build/sof_builder/Dockerfile (hunk 2 above)
Comment (fredoh9) [reply scripts/docker_build/sof_builder/Dockerfile:32]: jed is added

File: scripts/docker_build/sof_builder/Dockerfile (hunk 3 above)
Comment (fredoh9) [reply scripts/docker_build/sof_builder/Dockerfile:71]: no we don't have any issue link. This issue was found after upgrading to Ubuntu 20.04. I referred to build from the scratch (https://thesofproject.github.io/latest/getting_started/build-guide/build-from-scratch.html) Do you want me to add this to commit text?

File: scripts/docker_build/sof_qemu/Dockerfile (hunk 4 above)
Comment (fredoh9) [reply scripts/docker_build/sof_qemu/Dockerfile:21]: Removed for QEMU images. But for sharing same code between two images, I don't have much idea right now. I'd like to do with new issue.

Comment (marc-hb) [reply scripts/docker_build/sof_qemu/Dockerfile:21]: > But for sharing same code between two images, I don't have much idea right now

OK, some other time then.

File: scripts/docker_build/sof_builder/Dockerfile (hunk 3 above)
Comment (marc-hb) [reply scripts/docker_build/sof_builder/Dockerfile:71]: The docs are not good either. Maybe just copy the error message if you don't this into a comment?

File: scripts/docker_build/sof_builder/docker-build.sh [hunk 5]
```diff
@@ -2,9 +2,4 @@
 # SPDX-License-Identifier: BSD-3-Clause
 # Copyright(c) 2018 Intel Corporation. All rights reserved.
 
-if [ -f "/etc/apt/apt.conf" ]; then
-	cp /etc/apt/apt.conf ./
-else
-	touch apt.conf
-fi
-docker build --build-arg UID=$(id -u) --build-arg host_http_proxy=$http_proxy --build-arg host_https_proxy=$https_proxy -t sof .
+docker build --build-arg UID=$(id -u) --build-arg host_http_proxy="$http_proxy" --build-arg host_https_proxy="$https_proxy" -t sof .
```
Comment (marc-hb) [inline scripts/docker_build/sof_builder/docker-build.sh:5]: Is `host_http_proxy` still used anywhere?

Comment (fredoh9) [reply scripts/docker_build/sof_builder/docker-build.sh:5]: that's right!

File: scripts/docker_build/sof_builder/Dockerfile (hunk 3 above)
Comment (fredoh9) [reply scripts/docker_build/sof_builder/Dockerfile:71]: already commit text has oneline error message.

```
[  9%] Generating sof-cnl-rt274.tplg
alsatplg: error while loading shared libraries: libatopology.so.2: cannot open shared object file: No such file or directory
make[2]: *** [topology/CMakeFiles/topology_sof-cnl-rt274.dir/build.make:61: topology/sof-cnl-rt274.tplg] Error 127
```

by default, alsa libraries go to /usr/lib64.

Comment (marc-hb) [reply scripts/docker_build/sof_builder/Dockerfile:71]: I think this error deserves more visibility and to be mentioned not just in the commit message but in a comment too. It's just a comment so you don't have to re-test :-)

Testing changes here is very time-consuming, one reason why these configuration files need even more comments than usual IMHO.

Comment (fredoh9) [reply scripts/docker_build/sof_builder/Dockerfile:75]: yes, I agree. I added that as comments also.

**Review by marc-hb — APPROVED** [review]

**Comment by fredoh9** [comment]
@marc-hb good point. Sorry for missing that. Fixing this now

**Review by lgirdwood — APPROVED** [review]
